src/Ocelot/Configuration/Creator/UpstreamHeaderRoutingOptionsCreator.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Ocelot.Configuration.File;
+
+namespace Ocelot.Configuration.Creator
+{
+    public class UpstreamHeaderRoutingOptionsCreator : IUpstreamHeaderRoutingOptionsCreator
+    {
+        public UpstreamHeaderRoutingOptions Create(FileUpstreamHeaderRoutingOptions options)
+        {
+            UpstreamHeaderRoutingTriggerMode mode = UpstreamHeaderRoutingTriggerMode.Any;
+            if (options.TriggerOn.Length > 0)
+            {
+                mode = Enum.Parse<UpstreamHeaderRoutingTriggerMode>(options.TriggerOn, true);
+            }
+
+            Dictionary<string, HashSet<string>> headers = options.Headers.ToDictionary(
```

Comment:
I'll be honest -- I [b]hate[/b] `var`s. I hated `auto` in C++, hate it here as well. I guess I'll oblige since it's your codebase but just so we're clear.

Reply:
I am not interested in things you hate in C++. I care about consistent style of the C# code base. You must understand that you are contributing to a public and open source product and your code will be read by thousands of developers.
If you think that you can create much better style & code conventions than standard rules by Microsoft team, you are wrong. Microsoft team recommends to use `var`, otherwise it wasn't designed.
The reason why I've pointed your attention is long line. Long lines make the code difficult to read. `var` declaration makes lines shorter with focus on variable initialization by business logic. Code reviewers and developers should focus on business logic.
Read this doc to ensure that `var` is helpful and useful: [Implicitly typed local variables - C# Programming Guide | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/implicitly-typed-local-variables)

Reply:
Fair enough. Apologies for my confrontational attitude. I have seen discussions on where to use `var` and as I understood them, it's kind of a gray area, although there are some guidelines by outlined by Microsoft. Like I said, I will try to go over my code and replace concrete types with `var` where applicable. Btw. I am much more a fan of the newer `Type value = new()` syntax, but since it's a new thing, I guess it's `var` for the time being.

Reply:
✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1684/commits/6d29040db6570a01847ef596a2cdc583e146ad6c